src/plugins/formulas/formulas.js

```diff
@@ -197,64 +254,202 @@ export class Formulas extends BasePlugin {
       });
     }
 
+    this.engine = setupEngine(
+      this.#settings,
+      getEngineSettingsOverrides(this.hot.getSettings()),
+      this.hot.guid
+    );
+
+    if (!this.engine) {
+      this.disablePlugin();
+
+      return;
+    }
+
+    // HyperFormula events:
+    this.engine.on('valuesUpdated', (...args) => this.onHFvaluesUpdated(...args));
```

Comment:
I don't think we should clear `this.engine` on `disablePlugin`, because of this use case:
```js
formulas: {
    engine: hot1.getPlugin('formulas').engine, // hot1 might as well be disabled
}
```

As for 
> Should we disable plugin if we didn't enabled it?

Although the `super` call comes later, the check is currently done after assigning all the hooks, so `disablePlugin` clears those. But I can move the engine initialization part to the top of `enablePlugin` and just kill it if something goes wrong.

Reply:
If you think it's under control I'm fine here. Thanks for taking my comment under consideration :)